# rnaDualSeq

This is a package to house functions that help facilitate and visualize RNA sequences to help with dual RNA sequence experiments.

## Installation

Use the package manager [pip](https://pip.pypa.io/en/stable/) to install foobar.

```bash
require("devtools")
devtools::install_github("smnthpang/rnaDualSeq", build_vignettes = TRUE)
library("rnaDualSeq")
```

## Overview

```python

ls("package:rnaDualSeq")
data(package = "rnaDualSeq")
```

See vignettes for tutorial on package:
```bash
ls("package:<rnaDualSeq>")
data(package = "<rnaDualSeq>") # optional
browseVignettes("<rnaDualSeq>")
```
An overview of the package is illustrated below:

## Contributing
The author of this package is Samantha Pang.

## References 

Dinarvand, M., Kock, F., Al Mouiee, D., Vuong, K., Vijayan, A., Tanzim, A. F., Azad, A. K. M., Penesyan, A., Castaño-Rodríguez, N., &amp; Vafaee, F. (2022). DSEQSB: A Systems Biology Approach to decipher dynamics of host-pathogen interactions using temporal dual RNA-Seq Data. https://doi.org/10.1101/2022.02.28.482417 
Doyle, M. (2022, October 18). Galaxy training: Visualization of RNA-seq results with volcano plot in R. Galaxy Training Network. Retrieved November 15, 2022, from https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html 
Love, M. I., Anders, S., Kim, V., &amp; Huber, W. (n.d.). RNA-seq workflow: gene-level exploratory analysis and differential expression. RNA-seq workflow: Gene-level exploratory analysis and differential expression. Retrieved November 15, 2022, from https://www.bioconductor.org/packages/devel/workflows/vignettes/rnaseqGene/inst/doc/rnaseqGene.html 
Walch, P., Selkrig, J., Knodler, L. A., Rettel, M., Stein, F., Fernandez, K., Viéitez, C., Potel, C. M., Scholzen, K., Geyer, M., Rottner, K., Steele-Mortimer, O., Savitski, M. M., Holden, D. W., &amp; Typas, A. (2021). Global mapping of salmonella enterica-host protein-protein interactions during infection. Cell Host &amp; Microbe, 29(8). https://doi.org/10.1016/j.chom.2021.06.004 
Westermann, A. J., Förstner, K. U., Amman, F., Barquist, L., Chao, Y., Schulte, L. N., Müller, L., Reinhardt, R., Stadler, P. F., &amp; Vogel, J. (2016). Dual RNA-seq unveils noncoding RNA functions in host–pathogen interactions. Nature, 529(7587), 496–501. https://doi.org/10.1038/nature16547 
Macho Rendón, J., Lang, B., Ramos Llorens, M., Tartaglia, G.G., and Torrent Burgas, M. (2021). DualSeqDB: a database to assess the relevance of bacterial genes during host infection. Nucleic Acids Res. 49, D687–D693.
Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK (2015). “limma powers differential expression analyses for RNA-sequencing and microarray studies.” Nucleic Acids Research, 43(7), e47. doi: 10.1093/nar/gkv007.
Wickham H (2016). ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York. ISBN 978-3-319-24277-4, https://ggplot2.tidyverse.org.
Robinson MD, McCarthy DJ and Smyth GK (2010). edgeR: a
  Bioconductor package for differential expression analysis of
  digital gene expression data. Bioinformatics 26, 139-140
Robinson MD, McCarthy DJ and Smyth GK (2010). edgeR: a
  Bioconductor package for differential expression analysis of
  digital gene expression data. Bioinformatics 26, 139-140


## Acknowledgements 
This package was developed as part of an assessment for 2022
BCB410H: Applied Bioinformatics course at the University of
Toronto, Toronto, CANADA. <PackageName>welcomes issues,
enhancement requests, and other contributions. To submit an issue,
use the GitHub issues
